### PR TITLE
Fixes #33970 - Append custom kernel options for kickstart

### PR DESCRIPTION
**Pull request by jeffmcutter**
For unattended provisioning append parameterized custom kernel options for kickstart_kernel_options.erb snippet

My use case in question is for adding serial console options to the kernel options to be able to have console access during kickstart / discovery / etc., but any desired kernel options could also be added.

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
ok to test

File: app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb [hunk 1]
```diff
@@ -136,6 +136,10 @@ description: |
   elsif nic_delay
     options.push("nicdelay=#{nic_delay} linksleep=#{nic_delay}")
   end
+
+  if host_param('append_kernel_options')
```
Comment (ezr-ondrej) [inline app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb:140]: @lzap does the name of the param make sense?
Do we already document the params somewhere, where we should add it?

File: app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb [hunk 2]
```diff
@@ -136,6 +136,10 @@ description: |
...
+    options.push("#{host_param('append_kernel_options')}")
```
Comment (ares) [inline app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb]: you don't have to use the interpolation here, e.g. just `options.push(host_param('append_kernel_options'))` should work

**Comment by ares** [comment]
There's already a `pxe_kernel_options` parameter, which I think can be used. It's not used directly in the snippet, but it's used in 
- kickstart_default_pxelinux.erb
- kickstart_default_pxegrub.erb
- kickstart_default_ipxe.erb
- kickstart_default_pxegrub2.erb

just before the snippet is rendered.

**Comment by jeffmcutter** [comment]
> There's already a `pxe_kernel_options` parameter, which I think can be used. It's not used directly in the snippet, but it's used in
> 
> * kickstart_default_pxelinux.erb
> * kickstart_default_pxegrub.erb
> * kickstart_default_ipxe.erb
> * kickstart_default_pxegrub2.erb
> 
> just before the snippet is rendered.

How is the pxe_kernel_options parameter set?  I tried setting it as a parameter on the host and it doesn't seem to have any effect.  I removed it from my kickstart_default_ipxe.erb file and restarted and it seems to be an empty string when rendered.

Thanks,
-Jeff

**Comment by ezr-ondrej** [comment]
ok to test

**Comment by ezr-ondrej** [comment]
> How is the pxe_kernel_options parameter set? I tried setting it as a parameter on the host and it doesn't seem to have any effect. I removed it from my kickstart_default_ipxe.erb file and restarted and it seems to be an empty string when rendered.

@lzap  would you know what that param is for?

**Comment by jeffmcutter** [comment]
Is this waiting on contributor or something else?

File: app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb (hunk 1 above)
Comment (lzap) [reply app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb:140]: This snippet is included in PXE templates:

```
app/views/unattended/provisioning_templates/PXELinux/kickstart_default_pxelinux.erb
38:APPEND initrd=<%= @initrd %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>
```

As you can see, there is a helper called `pxe_kernel_options` defined in `app/models/operatingsystem.rb` which then takes kernelcmd host parameter. It is a bug tricky to see, but it is documented in our macros.

**Comment by lzap** [comment]
Sorry for the delay, please use `kernelcmd` host/hostgroup option and it will be automatically added into some PXE templates. If you want it in other PXE templates, then update those templates to call the macro:

```
app/views/unattended/provisioning_templates/iPXE/preseed_default_ipxe.erb
41:kernel <%= kernel %> initrd=initrd.img interface=auto url=<%= foreman_url('provision')%><%= provision_url_suffix %> ramdisk_size=10800 root=/dev/rd/0 rw auto BOOTIF=01-${netX/mac:hexhyp} hostname=<%= @host.name %> domain=<%= @host.domain %> <%= keyboard_params %> <%= pxe_kernel_options %> locale=<%= host_param('lang') || 'en_US' %> <%= netcfg_args %>

app/views/unattended/provisioning_templates/iPXE/kickstart_default_ipxe.erb
43:kernel <%= "#{@host.url_for_boot(:kernel)}" %> initrd=initrd.img ks=<%= ks %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options", variables: {ipxe_net: true}).strip %>

app/views/unattended/provisioning_templates/PXELinux/kickstart_default_pxelinux.erb
38:APPEND initrd=<%= @initrd %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>

app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb
41:  if host_param('pxe_kernel_options')
42:    options << host_param('pxe_kernel_options')

app/views/unattended/provisioning_templates/PXEGrub2/preseed_default_pxegrub2.erb
44:  if host_param('pxe_kernel_options')
45:    options << host_param('pxe_kernel_options')

app/views/unattended/provisioning_templates/PXEGrub/kickstart_default_pxegrub.erb
30:  kernel (nd)/../<%= @kernel %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>

app/views/unattended/provisioning_templates/PXEGrub2/kickstart_default_pxegrub2.erb
41:  <%= linuxcmd %> <%= @kernel %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>
56:  <%= linuxcmd %> (http,<%= proxy_host %>:<%= proxy_http_port %>)/httpboot/<%= @kernel %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>
65:  <%= linuxcmd %> (https,<%= proxy_host %>:<%= proxy_https_port %>)/httpboot/<%= @kernel %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options").strip %>
```

Feel free to reopen if it does not work like you expect. Sorry for the huge delay.

**Pull request closed without merging**